astropy/modeling/powerlaws.py

```diff
@@ -507,3 +507,115 @@ def input_units(self):
     def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
         return {'x_0': inputs_unit[self.inputs[0]],
                 'amplitude': outputs_unit[self.outputs[0]]}
+
+
+class Schechter1D(Fittable1DModel):
+    r"""
+    Schechter luminosity function (`Schechter 1976
+    <https://ui.adsabs.harvard.edu/abs/1976ApJ...203..297S/abstract>`_),
+    parameterized in terms of magnitudes.
+
+    Parameters
+    ----------
+    phi_star : float
+        The normalization factor in units of number density.
+
+    m_star : float
+        The characteristic magnitude where the power-law form of the
+        function cuts off. Must not have units.
+
+    alpha : float
+        The power law index, also known as the faint-end slope. Must not
+        have units.
+
+    See Also
+    --------
+    PowerLaw1D, ExponentialCutoffPowerLaw1D, BrokenPowerLaw1D
+
+    Notes
+    -----
+    Model formula (with :math:`\phi^{*}` for ``phi_star``, :math:`M^{*}`
+    for ``m_star``, and :math:`\alpha` for ``alpha``):
+
+    .. math::
+
+        n(M) \ dM = (0.4 \ln 10) \ \phi^{*} \
+            [{10^{0.4 (M^{*} - M)}}]^{\alpha + 1} \
+            \exp{[-10^{0.4 (M^{*} - M)}]} \ dM
+
+    ``phi_star`` is the normalization factor in units of number density.
+    ``m_star`` is the characteristic magnitude where the power-law form
+    of the function cuts off into the exponential form. ``alpha`` is
+    the power-law index, defining the faint-end slope of the luminosity
+    function.
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        from astropy.modeling.models import Schechter1D
+        import astropy.units as u
+        import matplotlib.pyplot as plt
+        import numpy as np
+
+        phi_star = 4.3e-4 * (u.Mpc ** -3)
+        m_star = -20.26
```

Comment:
Does this model support the magnitude units?

Reply:
No.  `10 ** <mag Quantity>` fails (gives an error), so `m_star` cannot have mag units.  I state this in the docs.